arrow-continuations/src/main/kotlin/arrow/continuations/generic/DelimContScope.kt

```diff
@@ -25,12 +25,12 @@ class DelimContScope<R>(val f: suspend DelimitedScope<R>.() -> R) : DelimitedSco
   /**
    * Variable used for polling the result after suspension happened.
    */
-  private val resultVar = atomic<R?>(null)
+  private val resultVar = atomic<Any?>(EMPTY_VALUE)
 
   /**
    * Variable for the next shift block to (partially) run, if it is empty that usually means we are done.
    */
-  private val nextShift = atomic<(suspend () -> R)?>(null)
+  private val nextShift = atomic<Any?>(EMPTY_VALUE)
```

Comment:
I don't actually think this is necessary here. Only the result var actually ever has values of type `R` and thus can lead to nested nulls, this one cannot.

Reply:
I fixed this in the meantime, you were right, I confused the nextShift with the result atomic. Thanks for pointing that out! 👍